src/features/applicationView/offerCard/editForm/EditForm.tsx

```diff
@@ -0,0 +1,134 @@
+  const { t, i18n } = useTranslation();
+  const { data, loading } = useQuery<ORDER_OPTIONAL_PRODUCTS>(ORDER_OPTIONAL_PRODUCTS_QUERY);
+  const [createOrderLine] = useMutation<CREATE_ORDER_LINE, CREATE_ORDER_LINE_VARS>(CREATE_ORDER_LINE_MUTATION);
+  const [deleteOrderLine] = useMutation<DELETE_ORDER_LINE, DELETE_ORDER_LINE_VARS>(DELETE_ORDER_LINE_MUTATION);
+
+  if (loading)
+    return (
+      <div className={styles.editForm}>
+        <LoadingSpinner isLoading={loading} />
+      </div>
+    );
+  if (!data?.additionalProducts) return <div className={styles.editForm}>{t('common.notification.noData')}</div>;
+
+  const initialValues = data.additionalProducts.edges.reduce<Values>((acc, edge) => {
+    if (!edge?.node) return acc;
+    const productNodeId = edge.node.id;
+
+    return {
+      ...acc,
+      [edge.node.id]: selectedProducts.find(({ productId }) => productId === productNodeId) ? true : false,
+    };
+  }, {});
+
+  const onSubmit = async (values: Values) => {
+    type CreateMutation = ReturnType<typeof createOrderLine>;
+    type DeleteMutation = ReturnType<typeof deleteOrderLine>;
+
+    const createMutations: CreateMutation[] = [];
+    const deleteMutations: DeleteMutation[] = [];
+
+    Object.keys(initialValues).forEach((key) => {
+      if (initialValues[key] === values[key]) return;
+      if (values[key] === true)
+        return createMutations.concat([createOrderLine({ variables: { input: { orderId, productId: key } } })]);
+      if (values[key] === false) {
+        const product = selectedProducts.find(({ productId }) => productId === key);
+        return product && deleteMutations.concat([deleteOrderLine({ variables: { input: { id: product.orderId } } })]);
+      }
+    });
+
+    await Promise.all<CreateMutation | DeleteMutation>([...createMutations, ...deleteMutations]);
+
+    handleSubmit();
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={(values) => onSubmit(values)}
+      validateOnBlur={false}
+      validateOnChange={false}
+    >
+      {({ values, errors, handleChange, handleSubmit }) => {
+        type Product = { id: string; label: string; price: number };
+
+        const products = data?.additionalProducts?.edges.reduce<Product[]>((acc, edge) => {
+          if (!edge?.node) return acc;
+
+          const product = {
+            id: edge.node.id,
+            label: `${t(getProductServiceTKey(edge.node.service))}, ${t(getPeriodTKey(edge.node.period))}`,
+            price: edge.node.priceValue,
+          };
+
+          return [...acc, product];
+        }, []);
+
+        const ProductsItems = products?.map(({ id, label, price }, i) => (
+          <ListItem key={i} className={styles.listItem}>
+            <Checkbox id={id} name={id} checked={values[id]} onChange={handleChange} label={label} />
+            <span>{formatPrice(price, i18n.language)}</span>
+          </ListItem>
+        ));
```

Comment:
`products` could be moved before the return, and `ProductItems` could be inlined below

Reply:
Good point! 
Fixed.